spark/interpreter/src/main/scala/org/apache/zeppelin/spark/SparkZeppelinContext.scala

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.spark
+
+import java.util
+
+import org.apache.spark.SparkContext
+import org.apache.zeppelin.annotation.ZeppelinApi
+import org.apache.zeppelin.display.AngularObjectWatcher
+import org.apache.zeppelin.display.ui.OptionInput.ParamOption
+import org.apache.zeppelin.interpreter.{BaseZeppelinContext, InterpreterContext, InterpreterHookRegistry}
+
+import scala.collection.{JavaConversions, Seq}
+
+
+/**
+  * ZeppelinContext for Spark
+  */
+class SparkZeppelinContext(val sc: SparkContext,
+                           val sparkShims: SparkShims,
+                           val hooks2: InterpreterHookRegistry,
+                           val maxResult2: Int) extends BaseZeppelinContext(hooks2, maxResult2) {
+
+  private val interpreterClassMap = Map(
+    ("spark", "org.apache.zeppelin.spark.SparkInterpreter"),
+    ("sql", "org.apache.zeppelin.spark.SparkSqlInterpreter"),
+    ("dep", "org.apache.zeppelin.spark.DepInterpreter"),
+    ("pyspark", "org.apache.zeppelin.spark.PySparkInterpreter")
+  )
```

Comment:
In favor of more idiomatic Scala:

    "spark" -> "org.apache.zeppelin.spark.SparkInterpreter",
    "sql" -> "org.apache.zeppelin.spark.SparkSqlInterpreter",
    "dep" -> "org.apache.zeppelin.spark.DepInterpreter",
    "pyspark" -> "org.apache.zeppelin.spark.PySparkInterpreter"

Reply:
Fixed